oauth2_provider/oauth2_validators.py

```diff
@@ -237,6 +239,57 @@ def validate_client_id(self, client_id, request, *args, **kwargs):
     def get_default_redirect_uri(self, client_id, request, *args, **kwargs):
         return request.client.default_redirect_uri
 
+    def _get_token_from_authentication_server(self, token):
```

Comment:
Instead of this, have the token and the url be passed as function arguments instead of fetching them from the settings (as you already do in the caller).